presto-main/src/test/java/com/facebook/presto/type/BenchmarkDecimalOperators.java

```diff
@@ -87,6 +89,68 @@
     private static final SqlParser SQL_PARSER = new SqlParser();
 
     @State(Thread)
+    public static class CastDoubleToDecimalBenchmarkState
+            extends BaseState
+    {
+        private static final int SCALE = 2;
+
+        @Param({"10", "35", "BIGINT"})
+        private String precision;
+
+        @Setup
+        public void setup()
+        {
+            addSymbol("d1", DOUBLE);
+
+            String expression;
+            if (precision.equals("BIGINT")) {
+                setDoubleMaxValue(Long.MAX_VALUE);
```

Comment:
It's a cap for maximal generated values for `DoubleType`. This benchmark perform casts from `double` to either `bigint` or `decimal(x,y)`, we have to ensure that double value is not exceeding maximal value of target type.